src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspace.SolutionCreator.cs

```diff
@@ -113,118 +113,81 @@ public async Task<Solution> CreateSolutionAsync(Checksum newSolutionChecksum, Ca
             }
 
             private async Task<Solution> UpdateProjectsAsync(
-                Solution solution, SolutionStateChecksums oldSolutionChecksums, SolutionStateChecksums newSolutionChecksums, CancellationToken cancellationToken)
+                Solution solution, SolutionStateChecksums newSolutionChecksums, CancellationToken cancellationToken)
             {
-                using var olds = SharedPools.Default<HashSet<Checksum>>().GetPooledObject();
-                using var news = SharedPools.Default<HashSet<Checksum>>().GetPooledObject();
```

Comment:
the use of diffs of checksums, along with actually going back and retrieving project states was very confusing, and became difficult to manage in the world where a solution could be partially mutated.  I moved to a simpler model on the right.

note: a lot of this was complex because our old syncing system just had checksums, and didn't pass around basic info like project-ids.